compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -19,6 +21,29 @@ pub type Bool = c_uint;
 pub const True: Bool = 1 as Bool;
 pub const False: Bool = 0 as Bool;
 
+/// Wrapper for a raw enum value returned from LLVM's C APIs.
+///
+/// For C enums returned by LLVM, it's risky to use a Rust enum as the return
+/// type, because it would be UB if a later version of LLVM adds a new enum
+/// value and returns it. Instead, return this raw wrapper, then convert to the
+/// Rust-side enum explicitly.
+#[repr(transparent)]
+pub struct RawEnum<T> {
+    value: u32,
+    _rust_side_type: PhantomData<fn() -> T>,
```

Comment:
It doesn't matter much, since `T` is almost certainly going to be static anyway. But I'll try to think of something to write here.

My reasoning is basically “we don't own a T, and don't consume a T, but we do produce a T”. So the phantom function pointer reflects that.

Reply:
Oh right. The reason I asked is because usually when I see `PhantomData<>` containing a non-trivial type, i.e. not simply `PhantomData<T>`, I immediately assume it is doing something special w.r.t. variances and such.

Reply:
IIRC, `PhantomData<T>` causes `T` to affect things like auto-traits and dropck, which is why it's often not the appropriate choice.

But it's hard to even think about what it would mean for `T` to vary in this case.

Reply:
Yeah. I suppose whatever we have here is fine.